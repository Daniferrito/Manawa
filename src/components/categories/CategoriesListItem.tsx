import React from 'react';
import TaskReduced from '../tasks/TaskReduced';
import { IonCard } from '@ionic/react';
import Task from '../../types/Task';
import Category from '../../types/Category';

interface Props {
  category?: Category;
}

const CategoriesListItem = (props: Props) => {
  const { category } = props;

  if (category === undefined) return <>{'Error'}</>;

  return (
    <IonCard className="container">
      <p className="title">{category.title}</p>
      {category.tasks.slice(0, 2).map((task: Task) => (
        <TaskReduced key={task.title} task={task} />
      ))}
    </IonCard>
  );
};

export default CategoriesListItem;
