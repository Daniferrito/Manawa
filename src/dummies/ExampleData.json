{
    "categories": [
        {
            "title": "Work",
            "tasks": [
                {
                    "title": "Write report",
                    "description": "This is a description",
                    "expirationDate": "2020-01-20T12:00:00Z"
                },
                {
                    "title": "Meeting with stakeholders",
                    "description": "This is a description",
                    "expirationDate": "2020-01-21T12:00:00Z"
                }
            ]
        },
        {
            "title": "Personal",
            "tasks": [
                {
                    "title": "Buy food",
                    "description": "Fill the fridge",
                    "expirationDate": "2020-01-24T18:00:00Z"
                }
            ]
        }
    ]
}