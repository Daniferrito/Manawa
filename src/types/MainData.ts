import Category from './Category';

export default class MainData {
  categories: Category[];
  constructor(categories: Category[]) {
    this.categories = categories;
  }
  static fromJson(json: any): MainData {
    return new MainData(json.categories.map((category: any) => Category.fromJson(category)));
  }
}
