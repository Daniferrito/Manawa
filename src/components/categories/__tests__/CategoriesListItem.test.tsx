import React from 'react';
import { shallow } from 'enzyme';
import CategoriesListItem from '../CategoriesListItem';

import ExampleData from '../../../dummies/ExampleData.json';
import MainData from '../../../types/MainData';

test('renders without crashing', () => {
  const categoriesList = shallow(
    <CategoriesListItem category={MainData.fromJson(ExampleData).categories[0]} />
  );
  expect(categoriesList).toMatchSnapshot();
});
test('renders without crashing2', () => {
  const categoriesList = shallow(<CategoriesListItem />);
  expect(categoriesList).toMatchSnapshot();
});
