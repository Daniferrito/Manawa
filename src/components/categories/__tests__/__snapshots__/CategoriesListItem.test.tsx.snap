// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<ForwardRef(IonCard)
  className="container"
>
  <p
    className="title"
  >
    Work
  </p>
  <TaskReduced
    key="Write report"
    task={
      Task {
        "description": "This is a description",
        "expirationDate": 2020-01-20T12:00:00.000Z,
        "title": "Write report",
      }
    }
  />
  <TaskReduced
    key="Meeting with stakeholders"
    task={
      Task {
        "description": "This is a description",
        "expirationDate": 2020-01-21T12:00:00.000Z,
        "title": "Meeting with stakeholders",
      }
    }
  />
</ForwardRef(IonCard)>
`;

exports[`renders without crashing2 1`] = `
<Fragment>
  Error
</Fragment>
`;
